python/cuspatial/cuspatial/core/binpreds/feature_intersects.py

```diff
@@ -155,6 +148,29 @@ def _preprocess(self, lhs, rhs):
         )
 
 
+class LineStringPolygonIntersects(IntersectsPredicateBase):
+    def _preprocess(self, lhs, rhs):
+        intersects = lhs._basic_intersects(rhs)
+        contains = rhs._basic_contains_any(lhs)
+        return intersects | contains
```

Comment:
Well, I thought that the class name makes it pretty clear that the lhs is LineString and the rhs is Polygon, for example. Renaming them introduces two complexities:

1. Now the class name has redundancy:
```
class LineStringPolygonIntersects(BinPred):
    def _preprocess(self, linestrings, polygons):
        intersects = linestrings._basic_intersects(polygons)
        contains = linestrings._basic_contains(polygons)
        return intersects | contains
```
Which seems to repeat linestring/polygon a lot, which for me may actually be reducing readability. I kind of like it, too. Still thinking about that one.
2. What to do about same-type predicates?
```
class PolygonIntersectsPolygon(BinPred):
    def _preprocess(self, lhs_polygons, rhs_polygons):
        intersects = lhs_polygons._basic_intersects(rhs_polygons)
        contains_rhs = rhs_polygons._basic_contains(lhs_polygons)
        contains_lhs = lhs_polygons._basic_contains(rhs_polygons)
        return intersects | contains_rhs | contains_lhs
```
This example decreases readability rather than increasing it, imho.